Panic compiling webpack

```
panic: Unhandled case in Node.Text: *ast.ElementAccessExpression

goroutine 1915 [running]:
github.com/microsoft/typescript-go/internal/ast.(*Node).Text(0x4002ee0de0?)
        D:/typescript-go/internal/ast/ast.go:293 +0x2ec
github.com/microsoft/typescript-go/internal/ast.GetElementOrPropertyAccessName(0x400c949938?)
        D:/typescript-go/internal/ast/utilities.go:1314 +0x24
github.com/microsoft/typescript-go/internal/checker.isModuleExportsAccessExpression(0x4002ee0de0?)
        D:/typescript-go/internal/checker/utilities.go:1636 +0xa4
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkTestingKnownTruthyType(0x400c4fb008, 0x4002ee0de0, 0x400cace000, 0x4002edfcc0)
        D:/typescript-go/internal/checker/checker.go:3578 +0x90
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkTestingKnownTruthyTypes(0x400c4fb008, 0x4002ee0de0?, 0x400cace000, 0x4002edfcc0)
        D:/typescript-go/internal/checker/checker.go:3566 +0x4c
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkTestingKnownTruthyCallableOrAwaitableOrEnumMemberType(...)
        D:/typescript-go/internal/checker/checker.go:3561
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkConditionalExpression(0x400c4fb008, 0x7ff72be079a0?, 0x0)
        D:/typescript-go/internal/checker/checker.go:10325 +0x6c
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionWorker(0x400c4fb008, 0x7ff72b922f28?, 0xc4fb008?)
        D:/typescript-go/internal/checker/checker.go:7320 +0x49c
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionEx(0x400c4fb008, 0x4002ee0e40, 0x0)
        D:/typescript-go/internal/checker/checker.go:7073 +0x50
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkParenthesizedExpression(0x400c4fb008, 0x7ff72b9ed72c?, 0x0)
        D:/typescript-go/internal/checker/checker.go:9609 +0x38
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionWorker(0x400c4fb008, 0x7ff72b915b8c?, 0x2bb86680?)
        D:/typescript-go/internal/checker/checker.go:7290 +0x378
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionEx(0x400c4fb008, 0x4002b95900, 0x0)
        D:/typescript-go/internal/checker/checker.go:7073 +0x50
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkExpressionCachedEx(0x400c4fb008, 0x4002b95900, 0xc949c38?)
        D:/typescript-go/internal/checker/checker.go:7042 +0xb0
github.com/microsoft/typescript-go/internal/checker.(*Checker).getReturnTypeFromBody(0x400c4fb008, 0x4002e8f9a0, 0x0)
        D:/typescript-go/internal/checker/checker.go:18622 +0x2a0
github.com/microsoft/typescript-go/internal/checker.(*Checker).getReturnTypeOfSignature(0x400c4fb008, 0x4012033580)
        D:/typescript-go/internal/checker/checker.go:18517 +0x148
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkFunctionExpressionOrObjectLiteralMethodDeferred(0x400c4fb008, 0x4002e8f9a0)
        D:/typescript-go/internal/checker/checker.go:9722 +0x9c
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkDeferredNode(0x400c4fb008, 0x7ff72be06da0?)
        D:/typescript-go/internal/checker/checker.go:2283 +0xcc
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkDeferredNodes-range1(...)
        D:/typescript-go/internal/checker/checker.go:2268
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkDeferredNodes.(*OrderedSet[...]).Keys.func1(...)
        D:/typescript-go/internal/collections/ordered_map.go:117
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkDeferredNodes(0x400c4fb008, 0x4003183208?)
        D:/typescript-go/internal/checker/checker.go:2264 +0xc0
github.com/microsoft/typescript-go/internal/checker.(*Checker).checkSourceFile(0x400c4fb008, {0x7ff72be013f8, 0x7ff72c2e36e0}, 0x4002297608)
        D:/typescript-go/internal/checker/checker.go:2071 +0x100
github.com/microsoft/typescript-go/internal/checker.(*Checker).CheckSourceFile(0x400c4fb008, {0x7ff72be013f8, 0x7ff72c2e36e0}, 0x4002297608)
        D:/typescript-go/internal/checker/checker.go:2059 +0x58
github.com/microsoft/typescript-go/internal/compiler.(*Program).CheckSourceFiles.func1-range1(0x400b823f38?)
        D:/typescript-go/internal/compiler/program.go:297 +0x44
github.com/microsoft/typescript-go/internal/compiler.(*checkerPool).Files.func1(0x400d10a450)
        D:/typescript-go/internal/compiler/checkerpool.go:82 +0x7c
github.com/microsoft/typescript-go/internal/compiler.(*Program).CheckSourceFiles.func1()
        D:/typescript-go/internal/compiler/program.go:296 +0xc4
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
        D:/typescript-go/internal/core/workgroup.go:39 +0x50
created by github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue in goroutine 1
        D:/typescript-go/internal/core/workgroup.go:37 +0x7c
```

--- a/internal/ast/utilities.go
+++ b/internal/ast/utilities.go
@@ -1292,26 +1292,20 @@ func IsThisParameter(node *Node) bool {
 
 // Does not handle signed numeric names like `a[+0]` - handling those would require handling prefix unary expressions
 // throughout late binding handling as well, which is awkward (but ultimately probably doable if there is demand)
-func GetElementOrPropertyAccessArgumentExpressionOrName(node *Node) *Node {
+func GetElementOrPropertyAccessName(node *Node) *Node {
 	switch node.Kind {
 	case KindPropertyAccessExpression:
-		return node.Name()
+		if IsIdentifier(node.Name()) {
+			return node.Name()
+		}
+		return nil
 	case KindElementAccessExpression:
-		arg := SkipParentheses(node.AsElementAccessExpression().ArgumentExpression)
-		if IsStringOrNumericLiteralLike(arg) {
+		if arg := SkipParentheses(node.AsElementAccessExpression().ArgumentExpression); IsStringOrNumericLiteralLike(arg) {
 			return arg
 		}
-		return node
+		return nil
 	}
-	panic("Unhandled case in GetElementOrPropertyAccessArgumentExpressionOrName")
-}
-
-func GetElementOrPropertyAccessName(node *Node) string {
-	name := GetElementOrPropertyAccessArgumentExpressionOrName(node)
-	if name == nil {
-		return ""
-	}
-	return name.Text()
+	panic("Unhandled case in GetElementOrPropertyAccessName")
 }
 
 func IsExpressionWithTypeArgumentsInClassExtendsClause(node *Node) bool {
@@ -1356,7 +1350,11 @@ func GetNonAssignedNameOfDeclaration(declaration *Node) *Node {
 		bin := declaration.AsBinaryExpression()
 		kind := GetAssignmentDeclarationKind(bin)
 		if kind == JSDeclarationKindProperty || kind == JSDeclarationKindThisProperty {
-			return GetElementOrPropertyAccessArgumentExpressionOrName(bin.Left)
+			if name := GetElementOrPropertyAccessName(bin.Left); name != nil {
+				return name
+			} else {
+				return bin.Left
+			}
 		}
 		return nil
 	case KindExportAssignment, KindJSExportAssignment:
@@ -1426,22 +1424,17 @@ func GetAssignmentDeclarationKind(bin *BinaryExpression) JSDeclarationKind {
 	if IsModuleExportsAccessExpression(bin.Left) {
 		return JSDeclarationKindModuleExports
 	} else if (IsModuleExportsAccessExpression(bin.Left.Expression()) || IsExportsIdentifier(bin.Left.Expression())) &&
-		hasJSBindableName(bin.Left) {
+		GetElementOrPropertyAccessName(bin.Left) != nil {
 		return JSDeclarationKindExportsProperty
 	}
 	if bin.Left.Expression().Kind == KindThisKeyword {
 		return JSDeclarationKindThisProperty
 	}
-	if bin.Left.Kind == KindPropertyAccessExpression && IsIdentifier(bin.Left.Expression()) && hasJSBindableName(bin.Left) ||
+	if bin.Left.Kind == KindPropertyAccessExpression && IsIdentifier(bin.Left.Expression()) && IsIdentifier(bin.Left.Name()) ||
 		bin.Left.Kind == KindElementAccessExpression && IsIdentifier(bin.Left.Expression()) {
 		return JSDeclarationKindProperty
 	}
 	return JSDeclarationKindNone
-}
-
-func hasJSBindableName(node *Node) bool {
-	name := GetElementOrPropertyAccessArgumentExpressionOrName(node)
-	return IsIdentifier(name) || IsStringLiteralLike(name)
 }
 
 /**
@@ -2679,9 +2672,12 @@ func IsVariableDeclarationInitializedToRequire(node *Node) bool {
 }
 
 func IsModuleExportsAccessExpression(node *Node) bool {
-	return (IsPropertyAccessExpression(node) || isLiteralLikeElementAccess(node)) &&
-		IsModuleIdentifier(node.Expression()) &&
-		GetElementOrPropertyAccessName(node) == "exports"
+	if IsAccessExpression(node) && IsModuleIdentifier(node.Expression()) {
+		if name := GetElementOrPropertyAccessName(node); name != nil {
+			return name.Text() == "exports"
+		}
+	}
+	return false
 }
 
 func isLiteralLikeElementAccess(node *Node) bool {

--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -3575,7 +3575,7 @@ func (c *Checker) checkTestingKnownTruthyType(condExpr *ast.Node, condType *Type
 	if ast.IsLogicalOrCoalescingBinaryExpression(condExpr) {
 		location = ast.SkipParentheses(condExpr.AsBinaryExpression().Right)
 	}
-	if isModuleExportsAccessExpression(location) {
+	if ast.IsModuleExportsAccessExpression(location) {
 		return
 	}
 	if ast.IsLogicalOrCoalescingBinaryExpression(location) {
@@ -27743,7 +27743,7 @@ func (c *Checker) canGetContextualTypeForAssignmentDeclaration(node *ast.Node) b
 				return false
 			}
 			// and now for one single case of object literal methods
-			name := ast.GetElementOrPropertyAccessArgumentExpressionOrName(left)
+			name := ast.GetElementOrPropertyAccessName(left)
 			if name == nil {
 				return false
 			} else {

--- a/internal/checker/utilities.go
+++ b/internal/checker/utilities.go
@@ -1632,10 +1632,6 @@ func minAndMax[T any](slice []T, getValue func(value T) int) (int, int) {
 	return minValue, maxValue
 }
 
-func isModuleExportsAccessExpression(node *ast.Node) bool {
-	return ast.IsAccessExpression(node) && ast.IsModuleIdentifier(node.Expression()) && ast.GetElementOrPropertyAccessName(node) == "exports"
-}
-
 func getNonModifierTokenRangeOfNode(node *ast.Node) core.TextRange {
 	pos := node.Pos()
 	if node.Modifiers() != nil {

--- a/internal/parser/reparser.go
+++ b/internal/parser/reparser.go
@@ -24,7 +24,7 @@ func (p *Parser) reparseCommonJS(node *ast.Node) {
 		nodes[0].Flags = ast.NodeFlagsReparsed
 		nodes[0].Loc = bin.Loc
 		// TODO: Name can sometimes be a string literal, so downstream code needs to handle this
-		export = p.factory.NewCommonJSExport(p.newModifierList(bin.Loc, nodes), ast.GetElementOrPropertyAccessArgumentExpressionOrName(bin.Left), bin.Right)
+		export = p.factory.NewCommonJSExport(p.newModifierList(bin.Loc, nodes), ast.GetElementOrPropertyAccessName(bin.Left), bin.Right)
 	}
 	if export != nil {
 		export.Flags = ast.NodeFlagsReparsed

--- a/testdata/baselines/reference/conformance/deepElementAccessExpressionInJS.errors.txt
+++ b/testdata/baselines/reference/conformance/deepElementAccessExpressionInJS.errors.txt
@@ -1,0 +1,24 @@
+elementAccessExpressionInJS.js(1,5): error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ exports: { [key: string]: any; }; }'.
+  No index signature with a parameter of type 'string' was found on type '{ exports: { [key: string]: any; }; }'.
+elementAccessExpressionInJS.js(3,32): error TS7006: Parameter 'index' implicitly has an 'any' type.
+
+
+==== declarations.d.ts (0 errors) ====
+    declare var module: {
+        exports: {
+            [key: string]: any;
+        };
+    }
+==== elementAccessExpressionInJS.js (2 errors) ====
+    if (module[calculatePropertyName(1)]) {
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ exports: { [key: string]: any; }; }'.
+!!! error TS7053:   No index signature with a parameter of type 'string' was found on type '{ exports: { [key: string]: any; }; }'.
+    }
+    function calculatePropertyName(index) {
+                                   ~~~~~
+!!! error TS7006: Parameter 'index' implicitly has an 'any' type.
+        // this would be some webpack index in real life
+        return `property${index}`;
+    }
+    

--- a/testdata/baselines/reference/conformance/deepElementAccessExpressionInJS.symbols
+++ b/testdata/baselines/reference/conformance/deepElementAccessExpressionInJS.symbols
@@ -1,0 +1,28 @@
+//// [tests/cases/conformance/salsa/deepElementAccessExpressionInJS.ts] ////
+
+=== declarations.d.ts ===
+declare var module: {
+>module : Symbol(module, Decl(declarations.d.ts, 0, 11))
+
+    exports: {
+>exports : Symbol(exports, Decl(declarations.d.ts, 0, 21))
+
+        [key: string]: any;
+>key : Symbol(key, Decl(declarations.d.ts, 2, 9))
+
+    };
+}
+=== elementAccessExpressionInJS.js ===
+if (module[calculatePropertyName(1)]) {
+>module : Symbol(module, Decl(declarations.d.ts, 0, 11))
+>calculatePropertyName : Symbol(calculatePropertyName, Decl(elementAccessExpressionInJS.js, 1, 1))
+}
+function calculatePropertyName(index) {
+>calculatePropertyName : Symbol(calculatePropertyName, Decl(elementAccessExpressionInJS.js, 1, 1))
+>index : Symbol(index, Decl(elementAccessExpressionInJS.js, 2, 31))
+
+    // this would be some webpack index in real life
+    return `property${index}`;
+>index : Symbol(index, Decl(elementAccessExpressionInJS.js, 2, 31))
+}
+

--- a/testdata/baselines/reference/conformance/deepElementAccessExpressionInJS.types
+++ b/testdata/baselines/reference/conformance/deepElementAccessExpressionInJS.types
@@ -1,0 +1,32 @@
+//// [tests/cases/conformance/salsa/deepElementAccessExpressionInJS.ts] ////
+
+=== declarations.d.ts ===
+declare var module: {
+>module : { exports: { [key: string]: any; }; }
+
+    exports: {
+>exports : { [key: string]: any; }
+
+        [key: string]: any;
+>key : string
+
+    };
+}
+=== elementAccessExpressionInJS.js ===
+if (module[calculatePropertyName(1)]) {
+>module[calculatePropertyName(1)] : any
+>module : { exports: { [key: string]: any; }; }
+>calculatePropertyName(1) : string
+>calculatePropertyName : (index: any) => string
+>1 : 1
+}
+function calculatePropertyName(index) {
+>calculatePropertyName : (index: any) => string
+>index : any
+
+    // this would be some webpack index in real life
+    return `property${index}`;
+>`property${index}` : string
+>index : any
+}
+

--- a/testdata/tests/cases/conformance/salsa/deepElementAccessExpressionInJS.ts
+++ b/testdata/tests/cases/conformance/salsa/deepElementAccessExpressionInJS.ts
@@ -1,0 +1,18 @@
+// @allowJs: true
+// @checkJs: true
+// @strict: true
+// @target: es6
+// @outDir: ./out
+// @filename: declarations.d.ts
+declare var module: {
+    exports: {
+        [key: string]: any;
+    };
+}
+// @filename: elementAccessExpressionInJS.js
+if (module[calculatePropertyName(1)]) {
+}
+function calculatePropertyName(index) {
+    // this would be some webpack index in real life
+    return `property${index}`;
+}

